src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -255,15 +256,33 @@ private async Task<IPackageSearchMetadata> GetPackageMetadataAsync(PackageIdenti
                 tasks.Add(_localRepository.GetPackageMetadataFromLocalSourceAsync(identity, cancellationToken));
             }
 
+            if (_globalLocalRepositories != null)
+            {
+                _globalLocalRepositories.ForEach(x =>
+                    tasks.Add(x.GetPackageMetadataFromLocalSourceAsync(identity, cancellationToken)));
+            }
+
             IEnumerable<IPackageSearchMetadata> completed = (await Task.WhenAll(tasks))
                 .Where(m => m != null);
 
-            IPackageSearchMetadata master = completed.FirstOrDefault(m => !string.IsNullOrEmpty(m.Summary))
+            IPackageSearchMetadata packageSearchMetadataBase = completed.FirstOrDefault(m => !string.IsNullOrEmpty(m.Summary))
                 ?? completed.FirstOrDefault()
                 ?? PackageSearchMetadataBuilder.FromIdentity(identity).Build();
 
-            return master.WithVersions(
-                asyncValueFactory: () => MergeVersionsAsync(identity, completed));
+            var clonedResult = packageSearchMetadataBase
+                  .WithVersions(() => MergeVersionsAsync(identity, completed)) as PackageSearchMetadataBuilder.ClonedPackageSearchMetadata;
+
+            if (string.IsNullOrWhiteSpace(clonedResult.ReadmeFileUrl))
```

Comment:
clonedResult can be null

Reply:
WithVersions always returns a new ClonedPackageSearchMetadata instance. I can add the null check if though